roachprod: replace prom setup retries with default

herkolategan opened the pull request:
This change removes `c.RepeatRun` in favour of using the existing `c.Run` with the `install.WithRetryFn` option.

Epic: None
Release Note: None

herkolategan committed:
roachprod: replace prom setup retries with default

This change removes c.RepeatRun in favour of using the
existing c.Run with the `install.WithRetryFn` option.

Epic: None
Release Note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/117321)
<!-- Reviewable:end -->

srosenberg commented on pkg/roachprod/prometheus/prometheus.go:
```diff
@@ -253,7 +253,7 @@ func Init(
 		// https://github.com/prometheus/node_exporter/pull/2311
 		// At time of writing, there hasn't been a release in over half a year.
-		if err := c.RepeatRun(ctx, l, l.Stdout, l.Stderr, cfg.NodeExporter,
+		if err := c.Run(ctx, l, l.Stdout, l.Stderr, install.OnNodes(cfg.NodeExporter).WithRetryFn(install.AlwaysRetry),
```
Does it literally mean it never fails to retry? Previously, it would max out after 10 attempts albeit with an exponential backoff.

herkolategan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/117321)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @renatolabs and @srosenberg)

___
*[`pkg/roachprod/prometheus/prometheus.go` line 256 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/117321#gh-1445265732:-NnmqlTGEbLDkFK9dzDb:b-lj921l) ([raw file](https://github.com/cockroachdb/cockroach/blob/1070fe69b1d3ecd7ad62adc50ba991ef5222eef1/pkg/roachprod/prometheus/prometheus.go#L256)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Does it literally mean it never fails to retry? Previously, it would max out after 10 attempts albeit with an exponential backoff.
</blockquote></details>

That name is a little ambiguous, so I'm updating it to `WithShouldRetryFn` to make it more explicit. It only determines if a function should be retried depending on the previous results, and does not dictate the attempts. So it will use the default retry options which at this point is 3, with backoff etc.

I did however note a bug now with disabling retries altogether that relied on the previous functional nature of the options. Which I'll make an issue for.


<!-- Sent from Reviewable.io -->

herkolategan committed:
roachtest: rename WithRetryFn to WithShouldRetryFn

Previously, the naming of `WithRetryFn` made it difficult to discern what the
retry function intends to do. Renaming this function to `WithShouldRetryFn`
should make it a bit clearer. Also updated the comment on an existing should
retry function to indicate the intent.

Epic: None
Release Note: None

srosenberg reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/117321)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @herkolategan)

___
*[`pkg/roachprod/prometheus/prometheus.go` line 256 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/117321#gh-1445265732:-Nnv0lMkCEty_h0qRjZw:b-xs0855) ([raw file](https://github.com/cockroachdb/cockroach/blob/1070fe69b1d3ecd7ad62adc50ba991ef5222eef1/pkg/roachprod/prometheus/prometheus.go#L256)):*
> That name is a little ambiguous,

Maybe renaming to `True` would rid the ambiguity since it's clear that it's just a predicate. Otherwise, looking at the call site,

```
WithRetryFn(install.AlwaysRetry)
```

may leave things to interpretation, i.e., does `AlwaysRetry` mean retry indefinitely. I believe that's what tripped during the initial pass.


<!-- Sent from Reviewable.io -->

herkolategan committed:
roachtest: rename AlwaysRetry predicate to AlwaysTrue

Epic: None
Release Note: None

herkolategan commented:
TFTRs!

bors r=srosenberg,renatolabs

herkolategan commented:
bors retry
